### PR TITLE
ISPN-13181 JDK 17 changes (Nashorn)

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13181
https://issues.redhat.com/browse/ISPN-13272

### CODE REVIEW

**Comment by tristantarrant** [comment]
I'll add JDK 17 to the CI slaves as soon as PSI comes back from the dead...

**Comment by tristantarrant** [comment]
https://ci.infinispan.org/job/InfinispanAlternateBuilds/job/InfinispanJDKLatest/ is running against `main `

**Review by wburns — COMMENTED** [review]
Changes look fine, however CI is still failing. https://ci.infinispan.org/job/InfinispanAlternateBuilds/job/InfinispanJDKLatest/80/

**Comment by tristantarrant** [comment]
Because that CI job is running against `main`. Let me redirect it

**Comment by ryanemerson** [comment]
It looks like Nashorn needs to be added to the client classpath  for tests. There are also various IT failures.

**Comment by ryanemerson** [comment]
Further changes required. 74 failures atm.

**Comment by tristantarrant** [comment]
Please ignore the `thirdparty` failures as they will be fixed when WF 25 is out (https://issues.redhat.com/browse/WFCORE-5406)

**Comment by tristantarrant** [comment]
This is good to go

**Review by ryanemerson — COMMENTED** [review]
A couple of minor things, but it should be ok to remove the Jenkins commit and merge if the normal CI run is fine.

**Comment by ryanemerson** [comment]
Also, I think [ISPN-13182](https://issues.redhat.com/browse/ISPN-13182) is the wrong Jira ID as that has already been resolved and relates to blockhound.

File: server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java [hunk 1]
```diff
@@ -191,6 +193,7 @@ public void testScriptUpload() {
 
    @Test
    public void testExecScripts() {
+      SkipJunit.skipSinceJDK(16);
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java:196]: Adding the nashorn dependencies will force us to run all these tests in Container mode, but we don't have a JDK 17 container yet.

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java [hunk 2]
```diff
@@ -423,4 +433,82 @@ private String serverPath() {
...
+            }
+
+            @Override
+            public FileVisitResult postVisitDirectory(Path dir, IOException exc) {
+               return FileVisitResult.CONTINUE;
+            }
+         });
+         tar.close();
+         DockerClientFactory.instance().client().copyArchiveToContainerCmd(containers[server].getContainerId())
+               //.withHostResource(path)
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java]: Fixed

File: server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanemerson) [reply server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java:196]: Ok. I've created [ISPN-13269](https://issues.redhat.com/browse/ISPN-13269) so this isn't missed.

**Comment by ryanemerson** [comment]
`org.infinispan.server.configuration.ServerOverlayConfigurationTest.testOverlay` Is failing.

Also, shouldn't the `*.thirdparty.*` tests only fail with JDK 17?

**Comment by tristantarrant** [comment]
I did not modify the Jenkinsfile yet

**Comment by ryanemerson** [comment]
> I did not modify the Jenkinsfile yet

Argh sorry, I should have explicitly checked that! I just saw that the number of commits had reduced and assumed.

**Comment by tristantarrant** [comment]
I have added a commit which fixes the NPE in `org.infinispan.server.configuration.ServerOverlayConfigurationTest.testOverlay`

**Comment by tristantarrant** [comment]
Once CI is done I'll remove the Jenkinsfile commit

**Comment by tristantarrant** [comment]
@ryanemerson I've repushed it without the Jenkinsfile change

**Comment by ryanemerson** [comment]
There's a `ClassNotFoundException` with ` org.infinispan.commons.test.skip.SkipJunit` that needs to be resolved.

**Comment by tristantarrant** [comment]
> There's a `ClassNotFoundException` with ` org.infinispan.commons.test.skip.SkipJunit` that needs to be resolved.

Fixed

**Comment by tristantarrant** [comment]
@ryanemerson lgtm

**Comment by tristantarrant** [comment]
Silly me: I had made a change to the default `infinispan.xml` file

**Comment by tristantarrant** [comment]
I created an extra Jira and squashed the NPE commits

**Comment by ryanemerson** [comment]
The four `UNKNOWN.ThreadLeakChecker` failures are also consistently failing on this PR and aren't present on main.

**Comment by tristantarrant** [comment]
The arquillian upgrade has changed the class of the ConsoleConsumer thread. I've fixed it

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Comment by gustavonalle** [comment]
There was a possible related failure in org.infinispan.server.functional.RollingUpgradeDynamicStoreCliIT.testRollingUpgrade

**Pull request merged**